src/Common/IdempotentStorage/PredisIdempotentStorage.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Gaming\Common\IdempotentStorage;
+
+use Predis\ClientInterface;
+
+/**
+ * @template T
+ * @implements IdempotentStorage<T>
+ */
+final class PredisIdempotentStorage implements IdempotentStorage
+{
+    public function __construct(
+        private readonly ClientInterface $predis,
+        private readonly int $expireTimeInSeconds
+    ) {
+    }
+
+    public function add(string $idempotencyKey, mixed $value): mixed
+    {
+        $result = $this->predis->setnx($idempotencyKey, serialize($value));
```

Comment:
All of this can be done within a single roundtrip, see [here](https://redis.io/commands/set/). However, `predis/predis` doesn't offer this capability yet and adding a custom command or Lua script isn't worth it. It can be changed later if that becomes a bottleneck.